src/Serialization/main.jl

```diff
@@ -1,9 +1,135 @@
+################################################################################
+# Description of the saving and loading mechanisms
+#
+# We require that any types serialized through OSCAR are registered using the
+# @register_serialization_type macro, like so
+
+# @register_serialization_type NewType "String Representation of type" uses_id uses_params
+
+# The last three arguments are optional and can arise in any order. Passing a String
+# arguement will override how the type is store as a string. The last two are boolean
+# flags. When setting uses_id the object will be stored as a reference and will be
+# referred to throughout the serialization using a UUID. This should typically only
+# be used for types that do not have a fixed normal form for example PolyRing and MPolyRing.
+# Using the uses_params flag will serialize the object with a more structured type
+# description which will make the serialization more efficient see the discussion on
```

Comment:
What happens otherwise?

This description makes me think "gee, if this option optimizes things, shouldn't I always use it???"

Reply:
otherwise the type information is just a string. Ultimately I do think it should be used whenever possible. I am still trying to figure out when that is exactly and maybe not even exposing this to the user but for now it is a tool for the people who understand the mechanism (1.5 people?) I agree needs a bit more work